src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -558,6 +559,26 @@ export default class AmalgamatingBusinesses extends Mixins(AmalgamationMixin, Co
     this.isAddingAmalgamatingForeignBusiness = false
   }
 
+  /** Sets the specified business as the new holding/primary business. */
+  async newHoldingPrimaryBusiness (business: AmalgamatingBusinessIF): Promise<void> {
+    try {
+      // show spinner since the network calls below can take a few seconds
+      this.$root.$emit('showSpinner', true)
+
+      // fetch the new holding/primary business' data and update the prepopulated data
+      // this will overwrite office addresses, directors, share structure and contact info
+      this.updatePrepopulatedData(business, true)
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log('Error setting new holding/primary business =', error)
+      this.snackbarText = 'Error marking new business.'
+      this.snackbar = true
+    } finally {
+      // hide spinner
+      this.$root.$emit('showSpinner', false)
```

Comment:
Would it be the same if this was on line 571 (same result)? Just asking here out of curiosity.

Reply:
No, it wouldn't. By having it here, it's called whether or not there was an error.

Reply:
If I put it on line 571 and updatePrepopulatedData() failed, the spinner would never be hidden.

Reply:
OK, got it. Thanks. Resolving.